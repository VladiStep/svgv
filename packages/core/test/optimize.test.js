const { transform } = require('../src/index.js');
const { fixture, fixtures } = require('./utils.js');

beforeAll(() => {
  jest.spyOn(process, 'cwd').mockReturnValue(fixtures);
});

afterAll(() => {
  jest.restoreAllMocks();
});

describe('optimize', () => {
  it('should use default config file', async () => {
    const result = await transform(await fixture('raw.svg'));
    expect(result).toMatchSnapshot('result');

    for (const configFile of [undefined, null]) {
      const svgo = { configFile };
      const actual = await transform(await fixture('raw.svg'), { svgo });
      expect(actual).toEqual(result);
    }
  });

  it('should not use default config file', async () => {
    const svgo = { configFile: false };
    const result = await transform(await fixture('raw.svg'), { svgo });
    expect(result).toMatchSnapshot('result');
  });

  it('should use custom config file', async () => {
    const svgo = { configFile: 'foo.config.js' };
    const result = await transform(await fixture('raw.svg'), { svgo });
    expect(result).toMatchSnapshot('result');
  });

  it('should throw error for non-existent config file', async () => {
    const svgo = { configFile: 'non-existent.config.js' };
    const result = transform(await fixture('raw.svg'), { svgo });
    await expect(result).rejects.toThrow();
  });
});
