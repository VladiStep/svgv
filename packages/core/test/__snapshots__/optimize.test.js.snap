// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optimize should not use default config file: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/></svg></template>
"
`;

exports[`optimize should use custom config file: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\">
    <circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/>
</svg>
</template>
"
`;

exports[`optimize should use default config file: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\">
  <circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/>
</svg>
</template>
"
`;
